astropy/coordinates/sky_coordinate.py

```diff
@@ -1575,7 +1581,24 @@ def radial_velocity_correction(self, kind='barycentric', obstime=None,
             gr = location.gravitational_redshift(obstime)
             # barycentric redshift according to eq 28 in Wright & Eastmann (2014),
             # neglecting Shapiro delay and effects of the star's own motion
-            zb = gamma_obs * (1 + targcart.dot(beta_obs)) / (1 + gr/speed_of_light) - 1
+            zb = gamma_obs * (1 + targcart.dot(beta_obs)) / (1 + gr/speed_of_light)
```

Comment:
Really minimal, but below you do `beta_obs.dot(targcart)`, which to me somehow is much clearer (I like dotting into a unit vector more than the other way around), so maybe you can swap the two here as well, i.e., `beta_obs.dot(targcart)`?